torchtext/datasets/text_classification.py

```diff
@@ -0,0 +1,410 @@
+import os
+import re
+import torch
+from torchtext.utils import download_from_url, extract_archive
+from torchtext.data.utils import generate_ngrams
+from torchtext.vocab import build_vocab
+import random
+from torchtext import data
+from tqdm import tqdm
+from torchtext.data import dataset
+from torchtext.data.iterator import generate_iterators
+
+
+def download(url, raw_folder, dataset_name):
+    """Download the dataset if it doesn't exist in processed_folder already."""
+
+    train_csv_path = os.path.join(raw_folder,
+                                  dataset_name + '_csv',
+                                  'train.csv')
+    test_csv_path = os.path.join(raw_folder,
+                                 dataset_name + '_csv',
+                                 'test.csv')
+    if os.path.exists(train_csv_path) and os.path.exists(test_csv_path):
+        return
+
+    os.makedirs(raw_folder)
+    filename = dataset_name + '_csv.tar.gz'
+    url = url
+    path = os.path.join(raw_folder, filename)
+    download_from_url(url, path)
+    extract_archive(path, raw_folder, remove_finished=True)
+
+    print('Dataset %s downloaded.' % dataset_name)
+
+
+def preprocess(raw_folder, processed_folder, dataset_name):
+    """Preprocess the csv files."""
+
+    def text_normalize(src_filepath, tgt_filepath):
```

Comment:
I think you can split this out as well. It seems useful in its own right. But instead of taking a filepath, it can take a string and return a string (or a list of strings).

Reply:
Sure thing. Will do it.